src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs

```diff
@@ -1868,5 +1870,439 @@ public static void Sort<T>(this Span<T> span, Comparison<T> comparison)
+                }
+
+                int paddingRequired = alignment - value.Length;
+                if (paddingRequired <= 0)
+                {
+                    // The value is as large or larger than the required amount of padding,
+                    // so just write the value.
+                    return AppendFormatted(value);
+                }
+
+                // Write the value along with the appropriate padding.
+                Debug.Assert(alignment > value.Length);
+                if (alignment <= _destination.Length - _pos)
+                {
+                    if (leftAlign)
+                    {
+                        value.CopyTo(_destination.Slice(_pos));
+                        _pos += value.Length;
+                        _destination.Slice(_pos, paddingRequired).Fill(' ');
+                        _pos += paddingRequired;
+                    }
+                    else
+                    {
+                        _destination.Slice(_pos, paddingRequired).Fill(' ');
+                        _pos += paddingRequired;
+                        value.CopyTo(_destination.Slice(_pos));
+                        _pos += value.Length;
+                    }
+
+                    return true;
+                }
+
+                return Fail();
+            }
+            #endregion
+
+            #region AppendFormatted string
+            /// <summary>Writes the specified value to the handler.</summary>
+            /// <param name="value">The value to write.</param>
+            public bool AppendFormatted(string? value)
+            {
+                // Fast-path for no custom formatter and a non-null string that fits in the current destination buffer.
+                if (!_hasCustomFormatter &&
+                    value is not null &&
+                    value.TryCopyTo(_destination.Slice(_pos)))
+                {
+                    _pos += value.Length;
+                    return true;
+                }
+
+                return AppendFormattedSlow(value);
+            }
+
+            /// <summary>Writes the specified value to the handler.</summary>
+            /// <param name="value">The value to write.</param>
+            /// <remarks>
+            /// Slow path to handle a custom formatter, potentially null value,
+            /// or a string that doesn't fit in the current buffer.
+            /// </remarks>
+            [MethodImpl(MethodImplOptions.NoInlining)]
+            private bool AppendFormattedSlow(string? value)
+            {
+                if (_hasCustomFormatter)
+                {
+                    return AppendCustomFormatter(value, format: null);
+                }
+
+                if (value is null)
+                {
+                    return true;
+                }
+
+                if (value.TryCopyTo(_destination.Slice(_pos)))
+                {
+                    _pos += value.Length;
+                    return true;
+                }
```

Comment:
Since AppendFormattedSlow is only called when _hasCustomFormatter was true (already returned), value was null (already returned) or TryCopyTo failed, we can be pretty certain the body of this if is unreachable, right?

I'd probably put a Debug.Fail in it to justify its uncovered state to someone looking at ccov.

Reply:
Good point.